test/IntegrationTests/StartupHookTests.cs

```diff
@@ -38,78 +37,80 @@ public StartupHookTests(ITestOutputHelper output)
     {
         SetEnvironmentVariable("OTEL_SERVICE_NAME", ServiceName);
         SetEnvironmentVariable("OTEL_DOTNET_AUTO_TRACES_ENABLED_INSTRUMENTATIONS", "HttpClient");
-        _expectations.Add(new WebServerSpanExpectation(ServiceName, null, "SayHello", "SayHello", null));
-        _expectations.Add(new WebServerSpanExpectation(ServiceName, null, "HTTP GET", "HTTP GET", null, "GET"));
     }
 
     [Fact]
     [Trait("Category", "EndToEnd")]
     public void SubmitsTraces()
     {
-        const int expectedSpanCount = 2;
-
         var spans = RunTestApplication(enableStartupHook: true);
 
-        AssertExpectedSpansReceived(expectedSpanCount, spans);
+        AssertAllSpansReceived(spans);
     }
 
     [Fact]
     [Trait("Category", "EndToEnd")]
-    public void InstrumentationNotAvailableWhenStartupHookIsNotEnabled()
+    public void WhenStartupHookIsNotEnabled()
     {
-        const int expectedSpanCount = 0;
-
         var spans = RunTestApplication(enableStartupHook: false);
 
-        AssertExpectedSpansReceived(expectedSpanCount, spans);
+#if NETFRAMEWORK
+        AssertAllSpansReceived(spans);
+#else
+        AssertNoSpansReceived(spans);
+#endif
     }
 
-    [Theory]
+    [Fact]
     [Trait("Category", "EndToEnd")]
-    [InlineData(true)]
-    [InlineData(false)]
-    public void ApplicationIsExcluded(bool excludeApplication)
+    public void ApplicationIsNotExcluded()
     {
-        string applicationNameToExclude = string.Empty;
-        int expectedSpanCount = 2;
-        if (excludeApplication)
-        {
-            applicationNameToExclude = EnvironmentHelper.FullTestApplicationName;
-            expectedSpanCount = 0;
-        }
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_EXCLUDE_PROCESSES", $"dotnet;dotnet.exe");
 
-        SetEnvironmentVariable("OTEL_DOTNET_AUTO_EXCLUDE_PROCESSES", $"dotnet,dotnet.exe,{applicationNameToExclude}");
+        var spans = RunTestApplication();
 
-        var spans = RunTestApplication(enableStartupHook: true);
-
-        AssertExpectedSpansReceived(expectedSpanCount, spans);
+        AssertAllSpansReceived(spans);
     }
 
-    private IImmutableList<IMockSpan> RunTestApplication(bool enableStartupHook)
+    [Fact]
+    [Trait("Category", "EndToEnd")]
+    public void ApplicationIsExcluded()
     {
-        int agentPort = TcpPortProvider.GetOpenPort();
+#if NETFRAMEWORK
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_EXCLUDE_PROCESSES", $"dotnet;dotnet.exe;{EnvironmentHelper.FullTestApplicationName};{EnvironmentHelper.FullTestApplicationName}.exe");
+#else
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_EXCLUDE_PROCESSES", $"dotnet,dotnet.exe,{EnvironmentHelper.FullTestApplicationName},{EnvironmentHelper.FullTestApplicationName}.exe");
+#endif
```

Comment:
I do not understand it...
Reference: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/890

Reply:
Yeah, this conditional seems strange. Perhaps the intention was `;` for Windows and `,` otherwise. Anyway, we should look and clarify.

IIRC in the past there was a filter for `MockZipkinCollector` this allowed to wait for the correct spans. If the filter was removed and we are simply counting spans (not checking name or attributes) this can pass tests that should fail.

Reply:
This is a different code path as far as I see

Reply:
Here is where the difference comes from https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/3d6d2bbfc20a82c9331aa8a16457904cb85b7506/src/OpenTelemetry.AutoInstrumentation.StartupHook/StartupHook.cs#L128 and this is different logic for .NET Framework vs .NET (Core)

> Regarding https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/890: the issue is that paths `,` is not a good choice on Windows.

Let's park it for another PR or issue. Here I will make the fixes to make it working as currently described in the docs.